pkg/storage/mongodb/mongodb.go

```diff
@@ -0,0 +1,177 @@
+type Database struct {
+	client          *mongo.Client
+	jokesCollection *mongo.Collection
+}
+
+// NewDatabase creating a new Database object.
+func NewDatabase() (*Database, error) {
+	var db Database
+
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	client, err := mongo.Connect(ctx, options.Client().ApplyURI("mongodb://localhost:27017"))
+	db.client = client
+	collection := client.Database(DBName).Collection(JokesCollectionName)
+	db.jokesCollection = collection
+	return &db, err
+}
+
+// GetJokes method returns a number of jokes given by skip and limit parameters and total amount of jokes.
+func (d *Database) GetJokes(ctx context.Context, skip, limit int) ([]models.Joke, int, error) {
+	amount, err := d.jokesCollection.CountDocuments(ctx, bson.M{})
+	if err != nil {
+		return []models.Joke{}, int(amount), err
+	}
+
+	options := options.Find()
+	options.SetSkip(int64(skip))
+	options.SetLimit(int64(limit))
+
+	cur, err := d.jokesCollection.Find(ctx, bson.M{}, options)
+	if err != nil {
+		return []models.Joke{}, int(amount), err
+	}
+	defer cur.Close(ctx)
+
+	result := []models.Joke{}
+
+	if err := cur.All(ctx, &result); err != nil {
+		return result, int(amount), err
+	}
+	if skip > int(amount) {
+		return []models.Joke{}, int(amount), nil
+	}
+	return result, int(amount), nil
+}
+
+// AddJoke method creating new joke.
+func (d *Database) AddJoke(ctx context.Context, title, body string, score int) (models.Joke, error) {
+	id := primitive.NewObjectID().Hex()
+	joke := models.NewJoke(id, title, body, score)
+
+	_, err := d.jokesCollection.InsertOne(ctx, joke)
+
+	return joke, err
+}
+
+// GetJokesByText returns a number of jokes, which contain the desired text, given by skip and limit parameters and total amount of found jokes.
+func (d *Database) GetJokesByText(ctx context.Context, skip, limit int, text string) ([]models.Joke, int, error) {
+	filter := bson.M{"$or": []interface{}{
+		bson.M{"body": primitive.Regex{Pattern: text, Options: "i"}},
+		bson.M{"title": primitive.Regex{Pattern: text, Options: "i"}},
+	}}
+
+	amount, err := d.jokesCollection.CountDocuments(ctx, filter)
+	if err != nil {
+		return []models.Joke{}, int(amount), err
+	}
+
+	options := options.Find()
+	options.SetSkip(int64(skip))
+	options.SetLimit(int64(limit))
+
+	result := []models.Joke{}
+
+	cur, err := d.jokesCollection.Find(ctx, filter, options)
+	if err != nil {
+		return result, int(amount), err
+	}
+	defer cur.Close(ctx)
+
+	if err := cur.All(ctx, &result); err != nil {
+		return result, int(amount), err
+	}
+	if skip > int(amount) {
+		return []models.Joke{}, int(amount), nil
+	}
+	return result, int(amount), nil
+}
+
+// GetJokeByID returns joke that has the same id.
+func (d *Database) GetJokeByID(ctx context.Context, id string) (models.Joke, error) {
+	filter := bson.M{"_id": id}
+
+	var joke models.Joke
+
+	err := d.jokesCollection.FindOne(ctx, filter).Decode(&joke)
+	if err == mongo.ErrNoDocuments {
```

Comment:
I believe you can simplify this 
`if err == mongo.ErrNoDocuments {
 		return joke, storage.ErrJokeNotFound
 	}
 return joke, err`